storage/queue.go

```diff
@@ -170,9 +173,12 @@ func newEntry(data tessera.Entry) *entry {
 //
 // This func must only be called once, and will cause any current or future callers of index()
 // to be given the values provided here.
-func (e *entry) assign(idx uint64, err error) {
+func (e *entry) notify(err error) {
 	e.c <- func() (uint64, error) {
-		return idx, err
+		if err != nil {
+			return 0, err
+		}
+		return *e.data.Index(), nil
```

Comment:
e.data.Index() could return nil

Reply:
You mean `Index` could be `nil`?
That would be a logic error on the storage implementation, I think - it must not have honoured the requirement to call `MarshalBundleData()` for that to happen.

I think a `panic` is the right thing to do in this case (can't reasonably continue - the storage implementation is broken), although I could perhaps make it panic earlier?

Reply:
panic SGTM